PocketKit/Sources/PocketKit/Tags/TagsFilterViewModel.swift

```diff
@@ -46,13 +48,24 @@ class TagsFilterViewModel: ObservableObject {
         } else {
             allTags.append(contentsOf: fetchedTags)
         }
+
+        let event = SnowplowEngagement(type: .general, value: nil)
+        let contexts: [Context] = [UIContext.home.screen, UIContext.myList.taggedChip]
+        tracker.track(event: event, contexts)
+
         return allTags
     }
 
     func selectTag(_ tag: SelectedTag) {
         selectedTag = tag
         if case .notTagged = tag {
-            // TODO: Track Analytics (IN-151)
+            let event = SnowplowEngagement(type: .general, value: nil)
+            let contexts: [Context] = [UIContext.home.screen, UIContext.myList.notTagged]
+            tracker.track(event: event, contexts)
+        } else {
+            let event = SnowplowEngagement(type: .general, value: nil)
+            let contexts: [Context] = [UIContext.home.screen, UIContext.myList.taggedChip]
+            tracker.track(event: event, contexts)
```

Comment:
style preference, maybe even break this out to its own function since we're creating the engagement event 3 times?
```
let tagContext = `tag == .notTagged` ? UIContext.myList.notTagged :  UIContext.myList.taggedChip
let event = SnowplowEngagement(type: .general, value: nil)
let contexts: [Context] = [UIContext.home.screen, tagContext]
tracker.track(event: event, contexts)
```